### PR TITLE
Use class var for tf graph & set scope

### LINKED ISSUE #4: Tensorflow Error causing model to fail on predict
@ptitzler and I are both consistent;t getting the following error on prediction:

```
[2018-07-27 22:57:58,333] ERROR in app: Exception on /model/predict [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1805, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1791, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.6/site-packages/flask_restplus/api.py", line 313, in wrapper
    resp = resource(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/flask/views.py", line 88, in view
    return self.dispatch_request(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/flask_restplus/resource.py", line 44, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/flask_restplus/marshalling.py", line 101, in wrapper
    resp = f(*args, **kwargs)
  File "/workspace/api/model.py", line 56, in post
    generated_text = self.model_wrapper.predict(seed_text, gen_chars)
  File "/workspace/core/backend.py", line 84, in predict
    preds = self.model.predict(x, verbose=0)[0]
  File "/usr/local/lib/python3.6/site-packages/keras/models.py", line 1025, in predict
    steps=steps)
  File "/usr/local/lib/python3.6/site-packages/keras/engine/training.py", line 1839, in predict
    self._make_predict_function()
  File "/usr/local/lib/python3.6/site-packages/keras/engine/training.py", line 1029, in _make_predict_function
    **kwargs)
  File "/usr/local/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py", line 2499, in function
    return Function(inputs, outputs, updates=updates, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py", line 2442, in __init__
    with tf.control_dependencies(self.outputs):
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 4604, in control_dependencies
    return get_default_graph().control_dependencies(control_inputs)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 4310, in control_dependencies
    c = self.as_graph_element(c)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3323, in as_graph_element
    return self._as_graph_element_locked(obj, allow_tensor, allow_operation)
  File "/usr/local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3402, in _as_graph_element_locked
    raise ValueError("Tensor %s is not an element of this graph." % obj)
ValueError: Tensor Tensor("activation_1/Softmax:0", shape=(?, 61), dtype=float32) is not an element of this graph.
```

### PR DESCRIPTION
**Pull request by MLnick**
Fixes #4

### CODE REVIEW

**Comment by MLnick** [comment]
cc @ptitzler @ajbozarth @djalova

**Comment by ajbozarth** [comment]
I don't know enough about tensor flow to understand the code, but looks code style looks ok to me

File: core/backend.py [hunk 1]
```diff
@@ -19,10 +20,11 @@ def __init__(self, path=DEFAULT_MODEL_PATH, model_file=DEFAULT_MODEL_FILE):
         logger.info('Loading model from: {}...'.format(path))
         model_path = '{}/{}'.format(path, model_file)
         clear_session()
-        self.model = models.load_model(model_path)
-        # this seems to be required to make Keras models play nicely with threads
-        self.model._make_predict_function()
```
[the remark this answers is not in the document]
Comment (MLnick) [reply core/backend.py]: Yeah I originally used it for that reason - without it the Flask app didn't run. But I tried with and without it and agree there doesn't seem any major difference and it still all works without that line

**Comment by djalova** [comment]
LGTM

**Pull request merged**